README.md

```diff
@@ -1 +1,253 @@
+  }
+}
+
+case class PredictionModel(
+  modelName: String,
+  modelVersion: SemVer,
+  score: Double
+)
+
+object FactTypes {
+  val DateOfBirth = FactType[LocalDate]("date_of_birth")
+  val HeightFt = FactType[FeetAndInches]("height_ft")
+  val WeightLbs = FactType[Int]("weight_lbs")
+  val Prediction = FactType[PredictionModel]("prediction")
+}
+
+object JoeSchmoe {
+  val height = FactTypes.HeightFt(FeetAndInches(5, 8))
+  val weight = FactTypes.WeightLbs(260)
+  val dateOfBirth = FactTypes.DateOfBirth(LocalDate.of(1980, 1, 1))
+  val predictWeightloss = FactTypes.Prediction(PredictionModel("weightloss", SemVer(2, 0, 1), 0.85))
+  val facts = Facts(height, weight, dateOfBirth, predictWeightloss)
+}
+
+val query: TerminalFactsExp = {
+  or(
+    and(
+      withType(FactTypes.WeightLbs).whereAnyFactValue(_ > 250),
+      withType(FactTypes.HeightFt).whereAnyFactValue(_ < FeetAndInches(6, 0)) 
+    ),
+    and(
+      withType(FactTypes.WeightLbs).whereAnyFactValue(_ > 300),
+      withType(FactTypes.HeightFt).whereAnyFactValue(_ >= FeetAndInches(6, 0)) 
+    ),
+    withType(FactTypes.Prediction).whereAnyFactValue { value =>
+      and(
+        value.at(_.select(_.modelName)) === "weightloss",
+        value.at(_.select(_.score)) > 0.8,
+        value.at(_.select(_.modelVersion)) >= SemVer(2, 0, 0)
+      ) 
+    }
+  )
+}
+val rs = evalWithFacts(JoeSchmoe.facts)(query)
+assert(rs.isTrue)
+assert(rs.matchingFacts == Facts(JoeSchmoe.height, JoeSchmoe.weight, JoeSchmoe.predictWeightloss))
+```
+
+#### DSL Operators
+
+| Operator | Sub-Expression Type | Return Type | Description |
+| -------- | ------------------- | ----------- | ----------- |
+| `and[T, A]` / `&&[T, A]` | `Exp[T, A]` | Builds an expression that returns a falsey `A` when any sub-expression returns a falsey `A` result |
+| `or[T, A]` / `||[T, A]` | `Exp[T, A]` | Builds an expression that returns a truthy `A` when any sub-expression returns a truthy `A` result |
+| `withType[T]` / `withTypeIn[T]` | N/A | WhereBuilder[T, ResultSet] | A `TerminalFactsExp` builder that filters facts to the provided type, collecting all facts that match the (eventually) provided `CondExp` |
+
+#### WhereBuilder Operators
+
+**TODO: Finish these operators once we solidify the interface**
+
+### FactTypeSets
+
+</details>
+
+## Implementation
+
+<details>
+
+In order to interpret expressions built by our embedded DSL, we wrap our `ExpAlg` algebraic data type (ADT)
+in a `FreeApplicative` data structure from the `cats-free` library. This allows us to use operations such
+as `.map` and `.foldMap` to produce new applicative contexts that can be evaluated efficiently.
+
+### Interpreters
+
+In order to go from `Exp[T, A]` to another type, such as a Json `Encoder[A]` or `Decoder[A]`, you just need
+to define a *natural transformation* (also known as a `FunctionK[F[_], G[_]]`) from `ExpAlg[T, *] ~> G`.
+
+Check out these [helpful slides on the library design](
```

Comment:
I think some more description can go here. Maybe some links to free applicative information here.